Sources/Actions/Profile/ThemeOptions.php

```diff
@@ -52,6 +52,14 @@ public function execute(): void
 			Profile::$member->loadCustomFields('theme');
 		}
 
+		if (empty(Utils::$modSettings['allow_no_censored'])) {
+			$k = array_find_key(Utils::$context['theme_options'], fn($v) => is_array($v) && $v['id'] === 'show_no_censored');
+
+			if ($k !== null) {
+				unset(Utils::$context['theme_options'][$k]);
+			}
+		}
+
```

Comment:
```suggestion
```

As said in #8833,  I don't think we should be unsetting the theme option. It is already the case that If the admin turns off the ability to disable the censor, then the censor will be applied and the theme option will be ignored no matter what its value is. But if the admin turns back on the ability to disable the censor, we should go back to whatever the user had chosen for themselves previously. We can't do that if we forcibly discard the user's chosen preference.